logging/log_config.go

```diff
@@ -13,17 +13,46 @@ type LoggerPatternConfig struct {
 }
 
 const (
+	// pattern matching on loggers.
 	validLoggerSectionName             = `[a-zA-Z0-9]+([_-]*[a-zA-Z0-9]+)*`
 	validLoggerSectionNameWithWildcard = `(` + validLoggerSectionName + `|\*)`
 	validLoggerSections                = validLoggerSectionName + `(\.` + validLoggerSectionName + `)*`
 	validLoggerSectionsWithWildcard    = validLoggerSectionNameWithWildcard + `(\.` + validLoggerSectionNameWithWildcard + `)*`
 	validLoggerName                    = `^` + validLoggerSectionsWithWildcard + `$`
+
+	// resource configurations.
+	validNamespacePattern       = `([\w-]+|\*)`
+	validResourceTypePattern    = `(service|component|\*)`
+	validResourceSubTypePattern = validNamespacePattern
+	validModelNamePattern       = validNamespacePattern
+	validTypeSubsectionPattern  = `(` + validResourceTypePattern + `:` + validResourceSubTypePattern + `|remote:)`
+	validResourcePattern        = `^rdk.` + validNamespacePattern + `:` + validTypeSubsectionPattern + `\/` + validModelNamePattern + `$`
 )
 
-var loggerPatternRegexp = regexp.MustCompile(validLoggerName)
+var (
+	loggerPatternRegexp   = regexp.MustCompile(validLoggerName)
+	resourcePatternRegexp = regexp.MustCompile(validResourcePattern)
+)
 
 func validatePattern(pattern string) bool {
-	return loggerPatternRegexp.MatchString(pattern)
+	return loggerPatternRegexp.MatchString(pattern) || resourcePatternRegexp.MatchString(pattern)
+}
+
+func buildRegexFromPattern(pattern string) string {
+	var matcher strings.Builder
+	matcher.WriteRune('^')
+	for _, ch := range pattern {
+		switch ch {
+		case '*':
+			matcher.WriteString(validLoggerSections)
+		case '.':
+			matcher.WriteString(`\.`)
+		default:
+			matcher.WriteRune(ch)
+		}
+	}
+	matcher.WriteRune('$')
+	return matcher.String()
 }
 
 func (lr *loggerRegistry) updateLoggerRegistry(logConfig []LoggerPatternConfig) error {
```

Comment:
[nit] Can we move this method to `logger_manager.go`? Sort of odd to define a method on `loggerRegistry` away from all the other methods.

Reply:
done